client/verta/verta/endpoint/nvidia_gpu.py

```diff
@@ -0,0 +1,40 @@
+from enum import Enum
+
+
+class NvidiaGPU:
+    NUM_GPUS_ERR_MSG = "`number_of_gpus` must be a number greater than 0"
+    MODEL_ERR_MSG = "`model` must be an instance of `verta.endpoint.NvidiaGPUModel`"
+
+    def __init__(self, number, model=None):
+        self._validate_number_of_gpus(number)
+        if model is not None:
+            self._validate_model(model)
+        self.number = number
+        self.model = model
+
+    def _validate_number_of_gpus(self, number_of_gpus):
+        if not isinstance(number_of_gpus, int):
+            raise TypeError(self.NUM_GPUS_ERR_MSG)
+        if number_of_gpus <= 0:
+            raise ValueError(self.NUM_GPUS_ERR_MSG)
+
+    def _validate_model(self, model):
+        if not isinstance(model, NvidiaGPUModel):
+            raise TypeError(self.MODEL_ERR_MSG)
+
```

Comment:
idk where this check should go, but something like

```suggestion
        if isinstance(model, str):
            NvidiaGPUModel(model)  # raises exception if `model` isn't a NvidiaGPUModel value

```